bevy/src/tests.rs

```diff
@@ -0,0 +1,240 @@
+};
+use mongodb::{
+    bson::{RawDocumentBuf, doc, rawdoc, spec::BinarySubtype},
+    options::GridFsBucketOptions,
+};
+
+const MONGODB_URI: LazyCell<String> = LazyCell::new(|| std::env::var("MONGODB_URI").unwrap());
+
+#[test]
+fn use_plugin() {
+    let rt = tokio::runtime::Runtime::new().unwrap();
+    let plugin = rt.block_on(async {
+        let client = mongodb::Client::with_uri_str(&*MONGODB_URI).await.unwrap();
+
+        MongodbAssetPlugin::new(&client).await
+    });
+
+    bevy::app::App::new().add_plugins(plugin).run();
+}
+
+static PNM_IMAGE_DATA: &[u8] = b"P1\n1 1\n0";
+
+#[test]
+fn load_image_document() {
+    let rt = tokio::runtime::Runtime::new().unwrap();
+    let plugin = rt.block_on(async {
+        let client = mongodb::Client::with_uri_str(&*MONGODB_URI).await.unwrap();
+        let db = client.database("bevy_test");
+
+        let doc = rawdoc! {
+            "name": "pixel.pbm",
+            "data": mongodb::bson::Binary {
+                subtype: BinarySubtype::Generic,
+                bytes: PNM_IMAGE_DATA.to_owned(),
+            },
+        };
+        let coll = db.collection::<RawDocumentBuf>("doc_images");
+        coll.drop().await.unwrap();
+        coll.insert_one(doc.clone()).await.unwrap();
+
+        let meta_coll = db.collection::<RawDocumentBuf>("doc_images.meta");
+        meta_coll.drop().await.unwrap();
+
+        MongodbAssetPlugin::new(&client).await
+    });
+
+    let result = test_load_image(plugin, "mongodb://document/bevy_test/doc_images/pixel.pbm");
+    assert!(result.is_ok(), "{:?}", result);
+}
+
+#[test]
+fn load_image_document_with_meta() {
+    let rt = tokio::runtime::Runtime::new().unwrap();
+    let plugin = rt.block_on(async {
+        let client = mongodb::Client::with_uri_str(&*MONGODB_URI).await.unwrap();
+        let db = client.database("bevy_test");
+
+        let doc = rawdoc! {
+            "name": "pixel.pbm",
+            "data": mongodb::bson::Binary {
+                subtype: BinarySubtype::Generic,
+                bytes: PNM_IMAGE_DATA.to_owned(),
+            },
+        };
+        let coll = db.collection::<RawDocumentBuf>("doc_images2");
+        coll.drop().await.unwrap();
+        coll.insert_one(doc).await.unwrap();
+
+        let meta =
+            bevy::asset::meta::AssetMeta::<(), ()>::new(bevy::asset::meta::AssetAction::Ignore);
+        let meta_doc = rawdoc! {
+            "name": "pixel.pbm",
+            "data": mongodb::bson::Binary {
+                subtype: BinarySubtype::Generic,
+                bytes: meta.serialize(),
+            },
+        };
+        let meta_coll = db.collection::<RawDocumentBuf>("doc_images2.meta");
+        meta_coll.drop().await.unwrap();
+        meta_coll.insert_one(meta_doc).await.unwrap();
+
+        MongodbAssetPlugin::new(&client).await
+    });
+
+    let result = test_load_image(plugin, "mongodb://document/bevy_test/doc_images2/pixel.pbm");
+    let Err(ImageLoadFailure::LoadFailed(inner)) = result else {
```

Comment:
`inner` here is an `Arc<AssetLoadError>` so we can't directly pattern match on the whole thing :(